data/blog/boosting-mobile-session-replay-performance-with-view-renderer-v2.md

```diff
@@ -0,0 +1,312 @@
+| Avg         | 25.42 ms                                                                   | 20.74 ms                                                             | -4.68 ms                           | -18.40 %                        |
+| p50         | 24.56 ms                                                                   | 19.84 ms                                                             | -4.72 ms                           | -19.20 %                        |
+| p75         | 27.34 ms                                                                   | 22.42 ms                                                             | -4.92 ms                           | -18.01 %                        |
+| p95         | 30.32 ms                                                                   | 24.66 ms                                                             | -5.66 ms                           | -18.67 %                        |
+| Max         | 32.58 ms                                                                   | 24.92 ms                                                             | -7.66 ms                           | -23.52 %                        |
+
+Looks like we can shave off another ~18-19% of the time on the main thread! Sounds too good to be true, right?
+
+Because it actually is too good to be true ☹️
+
+During testing we noticed that the rendering using `layer.render(in:)` is **incomplete**, in particular the icons used in the tab bar did not show up in the rendered screenshot at all.
+
+<div align="center">
+  <img src="/images/boosting-mobile-session-replay-performance-with-view-renderer-v2/render-comparison.png" alt="Render Failure"/>
+  _Tab bar icons missing when using layer.render(in:) approach_
+</div>
+
+The impact of the missing UI is not entirely clear and we could not pinpoint the exact cause for this (with unverified assumptions it is a side effect of [SF Symbols](https://developer.apple.com/sf-symbols/) and the way fonts are rendered). We decided against adopting this behavior as the new default render method in the foreseeable future and stick with `drawHierarchy` instead.
+
+As some of you might still prefer the even faster render times over completely rendered frames, you can still opt-in by configuring the option `options.sessionReplay.enableFastViewRenderering = true`.
+
+# The Numbers Speak For Themselves
+
+The optimization achieved substantial performance improvements, with the most dramatic gains on older hardware. Frame drops decreased from 9-10 frames per second to approximately 2 frames per second on iPhone 8 devices, representing a massive win!
+
+Main thread blocking time improved from 155 milliseconds to 25 milliseconds per screenshot, an **84% reduction** that brings Session Replay overhead well within acceptable performance budgets even on resource-constrained devices.
```

Comment:
We say "80%" above if I understand correctly, so this seems a _little_ inconsistent ("We have our first win and reduced the average time by ~125ms or ~80%! 🎉")